taskana-adapter-camunda-listener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java

```diff
@@ -131,14 +134,22 @@ private void insertCompleteOrDeleteEventIntoOutbox(
     String camundaSchema = null;
 
     try {
+      String taskState = TASK_STATE_COMPLETED;
+      if (delegateTask.getEventName().equals("delete")) {
+        if (delegateTask.getExecution().isCanceled()) {
+          taskState = TASK_STATE_CANCELLED;
+        } else {
+          taskState = TASK_STATE_TERMINATED;
+        }
+      }
 
-      String taskIdJson = "{\"id\":\"" + delegateTask.getId() + "\"}";
+      String referencedTaskJson = String.format("{\"id\":\"%s\",\"taskState\":\"%s\"}",
```

Comment:
If we don't use objectMapper this is not really a referencedTaskJson anymore. Maybe payloadJson or taskIdAndTaskStateJson is more fitting?

Reply:
I disagree. 
1. It is identical to the string that is produced by objectMapper if you instruct objectmapper not to serialize nulls
2. on the adapter side, this String is deserialized by ObjectMapper into a ReferencedTask.
Nevertheless, I renamed it to 'payload'....